src/services/facilityService.ts

```diff
@@ -109,125 +127,177 @@ export async function searchFacilities(filters: gqlTypes.FacilitySearchFilters =
+ * - If you want to link an existing HealthcareProfessional to a Facility, add the healthcareprofessionalId to the `healthcareProfessionalIds` array. 
+     Use the action to add or remove the association. If an id isn't in the list, no change will occur. 
  * @param facilityId The ID of the facility in the database.
  * @param fieldsToUpdate The values that should be updated. They will be created if they don't exist.
  * @returns The updated Facility.
  */
-export const updateFacility = async (facilityId: string, fieldsToUpdate: Partial<dbSchema.Facility>):
-    Promise<Result<dbSchema.Facility | null>> => {
+export const updateFacility = async (facilityId: string, fieldsToUpdate: Partial<gqlTypes.UpdateFacilityInput>):
+    Promise<Result<gqlTypes.Facility>> => {
     try {
-        const facilityRef = dbInstance.collection('facilities').doc(facilityId)
+        const validationResult = validateUpdateFacilityInput(fieldsToUpdate)
 
-        const snapshot = await facilityRef.get()
-
-        const facilityToUpdate = mapDbEntityTogqlEntity(snapshot.data() as DocumentData)
+        if (validationResult.hasErrors) {
+            return validationResult as Result<gqlTypes.Facility>
+        }
 
-        const updatedFacilityValues: dbSchema.Facility = {
-            ...facilityToUpdate,
-            ...fieldsToUpdate,
+        const facilityRef = dbInstance.collection('facilities').doc(facilityId)
+        const dbDocument = await facilityRef.get()
+        const dbFacilityToUpdate = dbDocument.data() as dbSchema.Facility
+        const updatedDbFacility: dbSchema.Facility = {
+            ...dbFacilityToUpdate,
+            //todo the partial update should happen right here
             updatedDate: new Date().toISOString()
         }
 
-        await facilityRef.set(updatedFacilityValues, { merge: true })
+        //TODO fix this: add/remove healthcareprofessionalIds based on the actions. 
+        //TODO update the associated healthcare professionals based on these changes. 
+
+        // Object.keys((fieldsToUpdate, currentKey) => {
+        // const key = fieldsToUpdate[currentKey] as keyof gqlTypes.UpdateFacilityInput
+        // (Object.keys(fieldsToUpdate) as (keyof typeof fieldsToUpdate)[]).forEach((currentKey, index) => {
+
+        // for (const currentKey in fieldsToUpdate) {
+        //     if (currentKey && currentKey in updatedDbFacility) {
+        //         updatedDbFacility[currentKey as keyof gqlTypes.UpdateFacilityInput] =
+        //             fieldsToUpdate[currentKey as keyof gqlTypes.UpdateFacilityInput]
+        //     }
+
+        //     if(currentKey === 'healthcareProfessionalIds' && fieldsToUpdate.healthcareProfessionalIds) {
+        //         fieldsToUpdate.healthcareProfessionalIds.forEach((relationship, index) => {
+        //             if(relationship.action === gqlTypes.AssociationAction.Add) {
```

Comment:
It would be better to create issues for this than to check in large blocks of commented out code.

Reply:
Yeah, I commented it out since I couldn't get it working in this PR but it will be in the next one. I can copy it out and add it separately